src/jaegertracing/samplers/RemotelyControlledSampler.cpp

```diff
@@ -42,14 +43,21 @@ class HTTPSamplingManager : public sampling_manager::thrift::SamplingManagerIf {
         : _serverURI(net::URI::parse(serverURL))
         , _logger(logger)
     {
-        net::Socket socket;
-        socket.open(AF_INET, SOCK_STREAM);
-        _serverAddr = socket.connect(serverURL);
+        try {
+            net::Socket socket;
+            socket.open(AF_INET, SOCK_STREAM);
+            _serverAddr = socket.connect(serverURL);
+        } catch (...) {
+            utils::ErrorUtil::logError(_logger, "cannot connect to socket");
+        }
     }
 
     void getSamplingStrategy(SamplingStrategyResponse& result,
                              const std::string& serviceName) override
     {
+        if (_serverAddr == net::IPAddress()) {
```

Comment:
hmm, not following. _serverAddr is set to net::IPAddress() if the client couldn't reach agent?

Reply:
Yes. `net::IPAddress()` is the default value for an empty IP and the address is only set upon successful connection. The reason for this is to avoid contacting the server if we couldn't connect on construction. I guess the real solution is not to start the looping thread at all, but that is another class that ends up calling this periodically.